ckanext/xloader/helpers.py

```diff
@@ -42,3 +44,80 @@ def is_resource_supported_by_xloader(res_dict, check_access=True):
     else:
         is_supported_url_type = True
     return (is_supported_format or is_datastore_active) and user_has_access and is_supported_url_type
+
+
+def xloader_badge(resource):
+    # type: (dict) -> str
+    """
+    Displays a custom badge for the status of Xloader and DataStore for the specified resource.
+    """
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.show_badges', True)):
+        return ''
+
+    if not XLoaderFormats.is_it_an_xloader_format(resource.get('format')):
+        # we only want to show badges for supported xloader formats
+        return ''
+
+    is_datastore_active = resource.get('datastore_active', False)
+
+    try:
+        xloader_job = toolkit.get_action("xloader_status")({'ignore_auth': True},
+                                                           {"resource_id": resource.get('id')})
+    except toolkit.ObjectNotFound:
+        xloader_job = {}
+
+    if xloader_job.get('status') == 'complete':
+        # the xloader task is complete, show datastore active or inactive.
+        # xloader will delete the datastore table at the beggining of the job run.
+        # so this will only be true if the job is fully finished.
+        status = 'active' if is_datastore_active else 'inactive'
+    elif xloader_job.get('status') in ['submitting', 'pending', 'running', 'running_but_viewable', 'error']:
+        # the job is running or pending or errored
+        # show the xloader status
+        status = xloader_job.get('status')
+        if status == 'running_but_viewable':
+            # treat running_but_viewable the same as running
+            status = 'running'
+        elif status == 'submitting':
+            # treat submitting the same as pending
+            status = 'pending'
+    else:
+        # we do not know what the status is
+        status = 'unknown'
+
+    messages = {
+        'pending': toolkit._('Data awaiting load to DataStore'),
+        'running': toolkit._('Loading data into DataStore'),
+        'complete': toolkit._('Data loaded into DataStore'),
+        'error': toolkit._('Failed to load data into DataStore'),
+        'active': toolkit._('Data available in DataStore'),
+        'inactive': toolkit._('Resource not active in DataStore'),
+        'unknown': toolkit._('DataStore status unknown'),
+    }
+    debug_level_statuses = ['complete', 'active', 'inactive', 'unknown']
+
+    if status in debug_level_statuses and not toolkit.asbool(toolkit.config.get('ckanext.xloader.debug_badges', False)):
```

Comment:
I would prefer to reverse this, checking a list of statuses that can be shown regardless, then if the current status isn't on that list, require the debug flag.

Reply:
@ThrawnCA okay lemme know if that is better (and if I even did what you actually asked for haha). Thanks!